### PR TITLE
Allow user to add existing mail domains created externally

### LINKED ISSUE #333: A user can create a mail domain

### LINKED ISSUE #334: A user can see the activation status of a mail domain

### LINKED ISSUE #336: A user can not create a mailbox for a domain if it is not enabled

### PR DESCRIPTION
**Pull request by daproclaima**
## Purpose

A user can add an externally created mail domain directly in the UI. 
This domain has a 'pending,' 'failed,' 'disabled,' or 'enabled' status. If the mail domain status is 'enabled,' it can be used.

## Proposal

- Show a form on the `/mail-domains/add` page, taking a single 'Domain name' input
- Redirect to `/mail-domains/<slug>` once created, and show mail domain status in the top left corner and left panel links
- disable mailbox creation button when mail domain status is not `enabled`
- also improve some links semantics for issue #302

closes #333, closes #334, closes #336

### CODE REVIEW

File: src/frontend/apps/e2e/__tests__/app-desk/mail-domains-create.spec.ts [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+  abilities: {
+    get: true,
+    patch: true,
+    put: true,
+    post: true,
+    delete: true,
+    manage_accesses: true,
+  },
+};
+
```
Comment (AntoLC) [inline src/frontend/apps/e2e/__tests__/app-desk/mail-domains-create.spec.ts:40]: You don't need to mock anymore. Better to do all your checks without mocking, like that you test that the backend works as well as expected.

File: src/frontend/apps/desk/src/features/mail-domains/components/CardCreateMailDomain.tsx [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+          $theme="info"
+          $variation="text"
+          $size="m"
+          $textAlign="justify"
+        >
+          {t(
+            'Please, wait for our services to validate your request once you submitted this form.\n' +
+              'Once the domain is validated, you will be able to assign mailboxes to it.',
+          )}
+        </Text>
```
[the remark this answers is not in the document]
Comment (AntoLC) [reply src/frontend/apps/desk/src/features/mail-domains/components/CardCreateMailDomain.tsx:126]: image

File: src/frontend/apps/desk/src/features/teams/teams-panel/components/PanelActions.tsx [hunk 3]
```diff
@@ -45,11 +45,13 @@ export const PanelActions = () => {
...
       <StyledLink href="/teams/create">
         <BoxButton
+          as="span"
+          $margin={{ all: 'auto' }}
           aria-label={t('Add a team')}
           $color={colorsTokens()['primary-600']}
           tabIndex={-1}
         >
-          <IconAdd width={30} height={30} aria-hidden="true" />
+          <IconAdd width={27} height={27} aria-hidden="true" />
```
Comment (daproclaima) [inline src/frontend/apps/desk/src/features/teams/teams-panel/components/PanelActions.tsx:54]: I also modified this component because what I did in the other one to fix the button semantic caused a position difference. With this modification, both buttons are displayed at the same place and with the good semantic.

File: src/frontend/apps/desk/src/features/mail-domains/components/panel/PanelItem.tsx [hunk 4]
```diff
@@ -11,17 +12,31 @@ interface MailDomainProps {
...
+      case 'failed':
+        return t('[failed]');
+    }
+  };
+
   const activeStyle = `
     border-right: 4px solid ${colorsTokens()['primary-600']};
     background: ${colorsTokens()['primary-400']};
-    span{
+    span:not(:last-child){
```
Comment (AntoLC) [inline src/frontend/apps/desk/src/features/mail-domains/components/panel/PanelItem.tsx:39]: I am not sure it is what you wanted to do, you loose the color effect on all your texts.

File: src/frontend/apps/desk/src/features/mail-domains/components/MailDomainsContent.tsx [hunk 5]
```diff
@@ -150,39 +146,118 @@ export function MailDomainsContent({ mailDomain }: { mailDomain: MailDomain }) {
...
+      )}
+    </Box>
+  );
+};
+
+const AlertStatus = ({
+  t,
+  status,
+}: {
+  t: TFunction<'translations'>;
```
Comment (AntoLC) [inline src/frontend/apps/desk/src/features/mail-domains/components/MailDomainsContent.tsx:203]: Don't pass `t` as an argument, use the `const { t } = useTranslation();` directly in your component.

File: src/frontend/apps/desk/src/features/mail-domains/components/MailDomainsContent.tsx [hunk 6]
```diff
@@ -150,39 +146,118 @@ export function MailDomainsContent({ mailDomain }: { mailDomain: MailDomain }) {
...
+          message: t(
+            'This domain name is deactivated. No new mailboxes can be created.',
+          ),
+        };
+      case 'failed':
+        // https://react.i18next.com/latest/trans-component#how-to-get-the-correct-translation-string
+        return {
+          variant: VariantType.ERROR,
+          message: (
+            <Trans t={t} i18nKey="mail-domain-status-error-message">
```
Comment (AntoLC) [inline src/frontend/apps/desk/src/features/mail-domains/components/MailDomainsContent.tsx:228]: I don't think you need to use `Trans` in this case, moreover you use `t` as well inside the `Trans` component, it is one or the other.

File: src/frontend/apps/desk/src/features/mail-domains/components/panel/PanelItem.tsx [hunk 7]
```diff
@@ -49,29 +66,44 @@ export const PanelMailDomains = ({ mailDomain }: MailDomainProps) => {
...
+              text-overflow: ellipsis !important;
             `}
-          >
-            {mailDomain.name}
+            >
+              {mailDomain.name}
+            </Text>
+          </Box>
+          <Text $size="s" $theme="greyscale">
+            {statusText}
```
Comment (AntoLC) [inline src/frontend/apps/desk/src/features/mail-domains/components/panel/PanelItem.tsx:106]: In french it seems broken, maybe it is ok.

File: src/frontend/apps/e2e/__tests__/app-desk/mail-domain.spec.ts
Comment (AntoLC) [inline src/frontend/apps/e2e/__tests__/app-desk/mail-domain.spec.ts:1]: Lot of testcase, big job, I have the feeling lot of tests are similar, it could maybe be a but more DRY.
It could be in another PR though.

File: src/frontend/apps/desk/src/features/mail-domains/components/panel/PanelItem.tsx (hunk 4 above)
Comment (daproclaima) [reply src/frontend/apps/desk/src/features/mail-domains/components/panel/PanelItem.tsx:39]: Indeed, I intended to use different colors for the mail domain status according to the status value. But we will eventually keep it grey. I bring back the former CSS rule

**Comment by daproclaima** [comment]
@AntoLC, you will see this ugly component hiding the mail domain name when it's too long to show its status. For the moment, we decided to leave it as is.
Screenshot 2024-08-14 at 16 51 39

**Comment by AntoLC** [comment]
> @AntoLC, you will see this ugly component hiding the mail domain name when it's too long to show its status. For the moment, we decided to leave it as is. Screenshot 2024-08-14 at 16 51 39

We have a case a bit similar in `Impress`, we did like that:
image

It can give you some idea.

**Review by AntoLC — APPROVED** [review]
✅ Good job !
